### PR TITLE
agent_run quest content and test fixes

### PR DESCRIPTION
**Pull request by greglarkin**
LEARNVM-493 #resolved #time 2h

### CODE REVIEW

File: tests/agent_run_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+describe "Task 8:", host: :agent do
```
Comment (kjhenner) [inline tests/agent_run_spec.rb]: These tasks should start at 1 for each quest--the task numbering in the content is currently incorrect. Apologies for the mistake there!

File: tests/agent_run_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+end
+
+describe "Task 9:", host: :localhost do
+  it 'Sign the agent certificate' do
+    file('/root/.bash_history')
+      .content
+      .should match /puppet cert list/
+    file('/root/.bash_history')
+      .content
+      .should match /puppet cert sign agent.puppet.vm/
```
Comment (kjhenner) [inline tests/agent_run_spec.rb]: I think there should be an artifact created by signing this certificate. If not, you could use serverspec's exec matcher to run 'puppet cert list' and check that agent.puppet.vm doesn't appear.

File: tests/agent_run_spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    file('/var/opt/lib/pe-puppet/lib/shared/pe_server_version.rb')
+      .should be_file
+  end
+end
+
+describe "Task 11:", host: :localhost do
+  it 'Check for site.pp classification' do
+    file('/etc/puppetlabs/code/environments/production/manifests/site.pp')
+      .content
+      .should match /node\s+'agent\.puppet\.vm'\s+\{.*?notify\s+\{\s+'Hello\s+Puppet!':\s+\}.*?\}/mi
```
Comment (kjhenner) [inline tests/agent_run_spec.rb]: The node's name in the node block can be wrapped in double-quotes, single-quotes, or bare.
`/node\s+(['"])?agent\.puppet\.vm\1?\s+\{.*?notify\s+\{\s+'Hello\s+Puppet!':\s+\}.*?\}/` should work.

File: tests/agent_run_spec.rb (hunk 1 above)
Comment (greglarkin) [inline tests/agent_run_spec.rb]: Ok, I'll renumber those.

File: tests/agent_run_spec.rb (hunk 2 above)
Comment (greglarkin) [inline tests/agent_run_spec.rb]: Good point - will check on that.

**Comment by greglarkin** [comment]
@kjhenner I updated the code based on your suggestions and ready to be reviewed now.

**Comment by kjhenner** [comment]
@greglarkin That matcher for the node definition still has the single quotes

**Comment by greglarkin** [comment]
@kjhenner Oops, forgot that one - will add it now.

File: tests/agent_run_spec.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    file('/var/opt/lib/pe-puppet/lib/shared/pe_server_version.rb')
+      .should be_file
+  end
+end
+
+describe "Task 4:", host: :localhost do
+  it 'Check for site.pp classification' do
+    file('/etc/puppetlabs/code/environments/production/manifests/site.pp')
+      .content
+      .should match /node\s+(['"])?agent\.puppet\.vm\1\s+\{.*?notify\s+\{\s+'Hello\s+Puppet!':\s+\}.*?\}/mi
```
Comment (greglarkin) [inline tests/agent_run_spec.rb]: @kjhenner Updated the regexp

**Pull request merged**